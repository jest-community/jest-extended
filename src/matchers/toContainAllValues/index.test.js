import matcher from './';

expect.extend(matcher);

const shallow = {
  hello: 'world',
  foo: 0,
  bar: false
};
const deep = {
  message: shallow,
  donald: 'duck'
};
const deepArray = {
  message: [shallow],
  donald: 'duck'
};

describe('.toContainAllValues', () => {
  it('passes when given object contains primitive values', () => {
    expect(shallow).toContainAllValues(['world', 0, false]);
  });

  it('passes when given object contains all values including objects', () => {
    expect(deep).toContainAllValues([{ hello: 'world', foo: 0, bar: false }, 'duck']);
  });

  it('passes when given object contains all values including arrays', () => {
    expect(deepArray).toContainAllValues(['duck', [{ hello: 'world', foo: 0, bar: false }]]);
  });

  it('fails when given object does not contain all primitive values', () => {
    expect(() => expect(shallow).toContainAllValues(['hello', 0, false])).toThrowErrorMatchingSnapshot();
    expect(() => expect(shallow).toContainAllValues(['world', 0])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object does not contain all values including objects', () => {
    expect(() => expect(deep).toContainAllValues([{ hello: 'world' }])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object does not contain all values including arrays', () => {
    expect(() => expect(deepArray).toContainAllValues(['duck', [{ hello: 'world' }]])).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toContainAllValues', () => {
  it('passes when given object does not contain all primitive values', () => {
    expect(shallow).not.toContainAllValues(['foo', 0]);
  });

  it('passes when given object does not contain all values including objects', () => {
    expect(deep).not.toContainAllValues(['duck', { foo: 'bar' }]);
  });

  it('passes when given object does not contain all values including arrays', () => {
    expect(deepArray).not.toContainAllValues(['duck', [{ foo: 'bar' }]]);
  });

  it('fails when given object contains primitive values', () => {
    expect(() => expect(shallow).not.toContainAllValues(['world', 0, false])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object contains all values including objects', () => {
    expect(() =>
      expect(deep).not.toContainAllValues([{ hello: 'world', foo: 0, bar: false }, 'duck'])
    ).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object contains all values including arrays', () => {
    expect(() =>
      expect(deepArray).not.toContainAllValues(['duck', [{ hello: 'world', foo: 0, bar: false }]])
    ).toThrowErrorMatchingSnapshot();
  });
});
