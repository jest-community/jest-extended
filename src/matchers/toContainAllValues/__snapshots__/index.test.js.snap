// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toContainAllValues fails when given object contains all values including arrays 1`] = `
"expect(received).not.toContainAllValues(expected)

Expected object to not contain all values:
  [\\"duck\\", [{\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}]]
Received:
  {\\"donald\\": \\"duck\\", \\"message\\": [{\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}]}"
`;

exports[`.not.toContainAllValues fails when given object contains all values including objects 1`] = `
"expect(received).not.toContainAllValues(expected)

Expected object to not contain all values:
  [{\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}, \\"duck\\"]
Received:
  {\\"donald\\": \\"duck\\", \\"message\\": {\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}}"
`;

exports[`.not.toContainAllValues fails when given object contains primitive values 1`] = `
"expect(received).not.toContainAllValues(expected)

Expected object to not contain all values:
  [\\"world\\", 0, false]
Received:
  {\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}"
`;

exports[`.toContainAllValues fails when given object does not contain all primitive values 1`] = `
"expect(received).toContainAllValues(expected)

Expected object to contain all values:
  [\\"hello\\", 0, false]
Received:
  {\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}"
`;

exports[`.toContainAllValues fails when given object does not contain all primitive values 2`] = `
"expect(received).toContainAllValues(expected)

Expected object to contain all values:
  [\\"world\\", 0]
Received:
  {\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}"
`;

exports[`.toContainAllValues fails when given object does not contain all values including arrays 1`] = `
"expect(received).toContainAllValues(expected)

Expected object to contain all values:
  [\\"duck\\", [{\\"hello\\": \\"world\\"}]]
Received:
  {\\"donald\\": \\"duck\\", \\"message\\": [{\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}]}"
`;

exports[`.toContainAllValues fails when given object does not contain all values including objects 1`] = `
"expect(received).toContainAllValues(expected)

Expected object to contain all values:
  [{\\"hello\\": \\"world\\"}]
Received:
  {\\"donald\\": \\"duck\\", \\"message\\": {\\"bar\\": false, \\"foo\\": 0, \\"hello\\": \\"world\\"}}"
`;
